doc/examples/edges/plot_self_overlapping_polygon.py

```diff
@@ -0,0 +1,60 @@
+"""
+==================================
+Subdivide self-overlapping polygons
+==================================
+
+This example shows how to divide a self-overlapping polygon into non
+self-overlapping sub polygons. Each sub-polygon can be drawn or analyzed by
+separate.
```

Comment:
```suggestion
separately.
```

Can you please include the reference at the end of this intro? Syntax goes like: https://github.com/scikit-image/scikit-image/blob/b8ce4d0c2c65684b131870abfe60228136108d4e/doc/examples/segmentation/plot_mask_slic.py#L19-L22